lib/ansible/modules/source_control/gitlab/gitlab_project_variable.py

```diff
@@ -167,13 +184,25 @@ def native_python_main(this_gitlab, purge, var_list, state):
     existing_variables = [x.get_id() for x in gitlab_keys]
 
     for key in var_list:
+
+        if isinstance(var_list[key], str):
+            value = var_list[key]
+            masked = False
+            protected = False
+        else:
```

Comment:
How about having another check whether `isinstance(var_list[key], dict)`, and if that's not the case, fail the module?

Reply:
(Potentially do this before actually changing something, to avoid the module dying in the middle of some modification.)

Reply:
yes, maybe that's the safest way.